storage/redis_cluster.go

```diff
@@ -490,18 +505,23 @@ func (r RedisCluster) DeleteKeys(keys []string) bool {
 
 // StartPubSubHandler will listen for a signal and run the callback for
 // every subscription and message event.
-func (r RedisCluster) StartPubSubHandler(channel string, callback func(interface{})) error {
-	if r.singleton() == nil {
+func (r *RedisCluster) StartPubSubHandler(channel string, callback func(interface{})) error {
```

Comment:
if I understand right we have only one call to this function - when tyk process gets started. There would be only one `currentMu.Lock()` ever, not clear why do we need it at this place. I'd look into other approach tbh